article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -235,3 +236,24 @@ object ArticlePicker {
     tier
   }
 }
+
+object TestGroups {
+  /*
+    This was introduced when DotcomRendering became DotcomRendering1 and DotcomRendering2
+    so that we could have 30% of the audience eligible for DCR rendering, a percentage between
+    20% and 50% that could not be achieved with a single test. The problem to solve was to make both
+    tests behave like one.
+   */
+  def combinator(group1: String, group2: String): String = {
+    // User is participant of the combined experiment if they are participant in either of the two base experiments,
+    // else are control if control in either of the two base experiments,
+    // else are not in the test.
+    (group1, group2) match {
+      case ("participant", _) => "participant"
+      case (_, "participant") => "participant"
+      case ("control", _) => "control"
+      case (_, "control") => "control"
+      case _ => "excluded"
```

Comment:
I'm not sure this matters because the test buckets shouldn't overlap (right?), but:

I think if they did then this would make you more likely to be a participant than in the control, because the 3rd and 4th cases here would never be reached for `(participant, control)` or `(control, participant)`.

Reply:
Actually you are perfectly right. I wrote the code knowing the two buckets do not overlap but that assumption could be broken in the future and the current code would not prevent against that :)